scripts/task/ExportDeliveryClassXmlResults.php

```diff
@@ -0,0 +1,181 @@
+            }
+            if (file_exists($path)) {
+                unlink($path);
+            }
+            $zipArchive = $this->instantiateZip($path, true);
+
+            $deliveryNumber = 0;
+            $executionAcrossDeliveriesNumber = 0;
+            foreach ($deliveryClass->getInstances(true) as $delivery) {
+                $deliveryNumber++;
+                $this->report->add(Report::createInfo(sprintf('Starting to process delivery "%s"', $delivery->getLabel())));
+
+                $executions = $this->getResultsService()->getResultsByDelivery($delivery);
+
+                $this->report->add(Report::createInfo(sprintf('%d results were found for delivery "%s"', count($executions), $delivery->getLabel())));
+
+                $executionNumber = 0;
+                foreach ($executions as $execution) {
+                    $executionNumber++;
+                    $executionAcrossDeliveriesNumber++;
+
+                    // flush the archive by creating the object ZipArchive from scratch every 100 iterations to
+                    // prevent ZipArchive from memory leaking (it builds up all the changes in memory that are not 'committed')
+                    if ($executionAcrossDeliveriesNumber % self::EXECUTION_ITERATIONS_TO_REINSTANTIATE_ZIP_OBJECT === 0) {
+                        $zipArchive->close();
+                        $zipArchive = $this->instantiateZip($path);
+                    }
+                    $xml = $this->getQtiResultsService()->getQtiResultXml($delivery->getUri(), $execution);
+                    $executionObj = $this->getServiceProxy()->getDeliveryExecution($execution);
+
+                    $userId = $executionObj->getUserIdentifier();
+                    $xmlfileName = \common_Utils::isUri($userId) ? explode('#', $userId)[1] : $userId;
+                    $xmlfileName .= '.xml';
+
+                    $zipArchive->addFromString('results/' . $delivery->getLabel() . '/' . $xmlfileName, $xml);
+                    $resultCount++;
+                }
+            }
+        } catch (\Throwable $e) {
+            if (isset($execution)) {
+                $this->report->add(Report::createFailure(sprintf('Exception thrown while processing execution "%s": %s', $execution, $e->getMessage())));
+            } else {
+                $this->report->add(Report::createFailure($e->getMessage()));
+            }
+        }
+        $this->report->add(Report::createSuccess(sprintf('Done! %d results for %d deliveries were exported. Please download the ZIP file at this path: %s', $resultCount, $deliveryNumber, $path)));
+        return $this->report;
+    }
+
+    /**
+     * @throws \Exception
+     */
+    private function instantiateZip($path, $create = false)
+    {
+        $flags = null;
+        if ($create) {
+            $flags = \ZipArchive::CREATE;
+        }
+
+        $zipArchive = new \ZipArchive();
+
+        $zipOpenResult = $zipArchive->open($path, $flags);
```

Comment:
I think it should look something like this (second param for $zipArchive->open cannot be null)
$zipOpenResult =  $create ? $zipArchive->open($path, \ZipArchive::CREATE) : $zipArchive->open($path);

Reply:
thanks, fixed